internal/daemonset/daemonset.go

```diff
@@ -53,10 +54,20 @@ func NewCreator(client client.Client, kernelLabel string, scheme *runtime.Scheme
 	}
 }
 
-func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
+func (dc *daemonSetGenerator) GarbageCollect(ctx context.Context, mod *kmmv1beta1.Module, existingDS []appsv1.DaemonSet, validKernels sets.Set[string]) ([]string, error) {
```

Comment:
I would add a new `daemonSetType` with possible values `daemonSetTypeModuleLoader` and `daemonsetTypeDevicePlugin` to be passed to this function and add re-orgenize this function as a switch-case.

Reply:
There is a common case for DS deletion (former version and no running pods), and a private case (invalid kernel version). The common case is valid for both ModuleLoader and Device Plugin , and private case is valid only for ModuleLoader. So, adding a switch case will result in going over all DS and creating to slices (one for DevicePlugin and one for ModuleLoader. I don't think this will improve the current code

Reply:
Or two distinct methods.

Reply:
added a more distinct functions instead of ifs